CMakePresets.json

```
@@ -0,0 +1,172 @@
+                "CMAKE_CXX_FLAGS": "-fno-omit-frame-pointer -fno-optimize-sibling-calls -gfull -O0"
+            },
+            "environment": {
+                "VCPKG_SUFFIX": ""
+            }
+        },
+        {
+            "name": "release-flags",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            },
+            "environment": {
+                "VCPKG_SUFFIX": "-rel"
+            }
+        },
+        {
+            "name": "arm64",
+            "hidden": true,
+            "environment": {
+                "VCPKG_ARCH": "arm64"
+            },
+            "architecture": {
+                "value": "arm64",
+                "strategy": "external"
+            }
+        },
+        {
+            "name": "x86_64",
+            "hidden": true,
+            "environment": {
+                "VCPKG_ARCH": "x64"
+            },
+            "architecture": {
+                "value": "x64",
+                "strategy": "external"
+            }
+        },
+        {
+            "name": "linux",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_C_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang",
+                "CMAKE_CXX_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang++"
+            },
+            "environment": {
+                "VCPKG_TARGET": "$env{VCPKG_ARCH}-linux"
+            },
+            "condition": {
+                "type": "equals",
+                "lhs": "Linux",
+                "rhs": "${hostSystemName}"
+            }
+        },
+        {
+            "name": "macos",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_C_COMPILER": "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang",
+                "CMAKE_CXX_COMPILER": "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
+            },
+            "environment": {
+                "VCPKG_TARGET": "$env{VCPKG_ARCH}-osx"
+            },
+            "condition": {
+                "type": "equals",
+                "lhs": "Darwin",
+                "rhs": "${hostSystemName}"
+            }
+        },
+        {
+            "name": "vcpkg-debug",
+            "hidden": true,
+            "inherits": ["debug-flags", "linux", "macos"],
+            "binaryDir": "$env{INSTALL_DIR}/build/remill",
+            "generator": "Ninja",
+            "cacheVariables": {
+                "VCPKG_TARGET_TRIPLET": "$env{VCPKG_TARGET}$env{VCPKG_SUFFIX}",
+                "VCPKG_ROOT": "$env{VCPKG_ROOT}",
+                "CMAKE_INSTALL_PREFIX": "$env{INSTALL_DIR}"
+            }
+        },
+        {
+            "name": "vcpkg-release",
+            "hidden": true,
+            "inherits": ["release-flags", "linux", "macos"],
```

Unfortunately, this sort of inheritance doesn't work on macOS with the conditional presets. It reads the `linux` preset first, finds that it's disabled and errors out. It works on macOS if you change the ordering like this though
```suggestion
            "inherits": ["release-flags", "macos", "linux"],
```
but then it won't work on Linux

---

Thanks for looking into this! I was able to repro and working on a workaround where VCPKG_TARGET_TRIPLET is manually specified. I am still thinking about the best way to deal with VCPKG_TARGET_TRIPLET; probably will add some brains to the script to try and do the right thing.

---

Ok, I think I have a workaround that works for the common case (release vcpkg) but lets people handle a debug vcpkg build. Will push soon.